File: 3.0/golang/31.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func dfs(graph [][]int, visited map[int]bool, now int) {
	visited[now] = true
	for _, neig := range graph[now] {
		if visited[neig] == false {
			dfs(graph, visited, neig)
		}
	}
}

func main() {
	var N, M int

	in := bufio.NewReader(os.Stdin)
	fmt.Fscanf(in, "%d %d\n", &N, &M)

	graph := make([][]int, N+1)
	visited := make(map[int]bool)
	for i := 0; i < N+1; i++ {
		visited[i] = false
	}

	var v, e int
	for i := 0; i < M; i++ {
		fmt.Fscanf(in, "%d %d\n", &v, &e)
		graph[v] = append(graph[v], e)
		graph[e] = append(graph[e], v)
	}

	dfs(graph, visited, 1)

	var ans []int
	for i := 0; i < N+1; i++ {
		if visited[i] {
			ans = append(ans, i)
		}
	}

	fmt.Println(len(ans))
	for _, v := range ans {
		fmt.Print(v, " ")
	}
}

File: 3.0/python/31.py
from sys import setrecursionlimit
setrecursionlimit(10000)

N, M = map(int, input().split())
graph = [[] for _ in range(N+1)]
visited = {i: False for i in range(len(graph))}

for _ in range(M):
    v, e = map(int, input().split())
    graph[v].append(e)
    graph[e].append(v)

def dfs(graph, visited, now):
    visited[now] = True
    for neig in graph[now]:
        if not visited[neig]:
            dfs(graph, visited, neig)

dfs(graph, visited, 1)

ans = []
for k, v in visited.items():
    if v:
        ans.append(k)

print(len(ans))
print(*ans)